# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../php/libgd-symbol-visibility.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

If external PHP extension that require libgd symbols (gdchart et al.) should
have an chance to work we need to export it's symbols:

gdchart.so: undefined symbol: php_gd_gdFontMediumBold in Unknown on line 0

  - Rene Rebe <rene@exactcode.de>

--- php-5.3.2/ext/gd/libgd/gdfontmb.h.vanilla	2010-05-22 18:06:40.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gdfontmb.h	2010-05-22 18:06:48.000000000 +0200
@@ -18,8 +18,10 @@
 
 #include "gd.h"
 
+#pragma GCC visibility push(default)
 extern gdFontPtr gdFontMediumBold;
 extern gdFontPtr gdFontGetMediumBold(void);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 }
--- php-5.3.2/ext/gd/libgd/gd.h.vanilla	2010-05-22 17:47:50.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gd.h	2010-05-22 17:49:27.000000000 +0200
@@ -11,6 +11,8 @@
 
 #include "php_compat.h"
 
+#pragma GCC visibility push(default)
+
 #define GD_MAJOR_VERSION 2
 #define GD_MINOR_VERSION 0
 #define GD_RELEASE_VERSION 35
@@ -697,4 +699,6 @@
 /* 2.0.12: this now checks the clipping rectangle */
 #define gdImageBoundsSafe(im, x, y) (!((((y) < (im)->cy1) || ((y) > (im)->cy2)) || (((x) < (im)->cx1) || ((x) > (im)->cx2))))
 
+#pragma GCC visibility pop
+
 #endif /* GD_H */
--- php-5.3.2/ext/gd/libgd/gdfontg.h.vanilla	2010-05-22 18:05:30.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gdfontg.h	2010-05-22 18:06:02.000000000 +0200
@@ -19,8 +19,10 @@
 
 #include "gd.h"
 
+#pragma GCC visibility push(default)
 extern gdFontPtr gdFontGiant;
 extern gdFontPtr gdFontGetGiant(void);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 }
--- php-5.3.2/ext/gd/libgd/gdfonts.h.vanilla	2010-05-22 18:06:56.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gdfonts.h	2010-05-22 18:07:03.000000000 +0200
@@ -18,8 +18,10 @@
 
 #include "gd.h"
 
+#pragma GCC visibility push(default)
 extern gdFontPtr gdFontSmall;
 extern gdFontPtr gdFontGetSmall(void);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 }
--- php-5.3.2/ext/gd/libgd/gdfontt.h.vanilla	2010-05-22 18:07:09.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gdfontt.h	2010-05-22 18:07:17.000000000 +0200
@@ -19,8 +19,10 @@
 
 #include "gd.h"
 
+#pragma GCC visibility push(default)
 extern gdFontPtr gdFontTiny;
 extern gdFontPtr gdFontGetTiny(void);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 }
--- php-5.3.2/ext/gd/libgd/gdfontl.h.vanilla	2010-05-22 18:06:17.000000000 +0200
+++ php-5.3.2/ext/gd/libgd/gdfontl.h	2010-05-22 18:06:33.000000000 +0200
@@ -20,8 +20,10 @@
 
 #include "gd.h"
 
+#pragma GCC visibility push(default)
 extern gdFontPtr gdFontLarge;
 extern gdFontPtr gdFontGetLarge(void);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 }
